### ISSUE TITLE
rose config-edit: advanced search

### ISSUE THREAD

**Issue by arjclark** [metomi/rose#35]
We need an advanced search dialog for the config editor that can drive searches by metadata and combinations of fields - e.g. find all integers with 'radiation' in their descriptions.

It would also be useful for large suites to have the option to only search a single configuration.

We should also consider whether to make the default search case sensitive or not.

**Comment by hjoliver** [comment]
I agree with @jmancell. If you're not sure exactly what you're looking for (e.g. I want to see all config items related to "gravity wave drag", or whatever) then a metadata search - preferably including the full 'help' section - would be best.

**Comment by scwhitehouse** [comment]
+1 from me!
